src/namerd.rs

```diff
@@ -52,10 +52,7 @@ pub fn resolve<C>(addr: net::SocketAddr,
     where C: Connect
 {
     let url = {
-        let base = format!("http://{}:{}/api/1/resolve/{}",
-                           addr.ip(),
-                           addr.port().to_string(),
-                           namespace);
+        let base = format!("{}/api/1/resolve/{}", base_url, namespace);
         Url::parse_with_params(&base, &[("path", &target)]).unwrap()
```

Comment:
unwrap() will panic if the Url can't parse. For improved validation, switch to using .expect(message)